projects/go-lib/src/lib/components/go-table/go-table.component.ts

```diff
@@ -170,39 +216,124 @@ export class GoTableComponent implements OnInit, OnChanges {
+          this.selectAllCheckbox.nativeElement.indeterminate = false;
+        }
+      } else {
+        this.targetedRows.push(row);
+        this.selectAllCheckbox.nativeElement.indeterminate = true;
+      }
+    } else {
+      if (event.target.checked && index < 0) {
+        this.targetedRows.push(row);
+      } else {
+        this.targetedRows.splice(index, 1);
+      }
+    }
+
+    this.rowSelectionEvent.emit({
+      currentRow: {
+        data: row,
+        selected: event.target.checked
+      },
+      deselectedRows: this.selectAllChecked ? this.targetedRows : [],
+      selectionMode: this.determineSelectionMode(),
+      selectedRows: !this.selectAllChecked ? this.targetedRows : []
+     });
+  }
+
+  isRowSelected(row: any): boolean {
+    if (this.selectAllChecked && !this.isRowInTargeted(row)) {
+      return true;
+    } else if (!this.selectAllChecked && this.isRowInTargeted(row)) {
+      return true;
+    } else {
+      return false;
+    }
+  }
+
+  //#region Private Methods
+  private handleSort(): void {
+    const { sortConfig, sortable, tableData }:
+    {
+      sortConfig?: GoTableSortConfig,
+      sortable: boolean,
+      tableData: any[]
+    } = this.localTableConfig;
 
     if (sortConfig && sortable && tableData && sortConfig.column) {
       this.localTableConfig.tableData.sort(sortBy(sortConfig.column, Boolean(sortConfig.direction)));
     }
   }
 
-  private toggleSortDir(currDir: SortDirection) : SortDirection {
+  private toggleSortDir(currDir: SortDirection): SortDirection {
     return currDir === SortDirection.ascending ? SortDirection.descending : SortDirection.ascending;
   }
 
-  private sortClasses(columnField: string, direction: SortDirection) : boolean {
-    const { sortConfig } = this.localTableConfig;
+  private sortClasses(columnField: string, direction: SortDirection): boolean {
+    const { sortConfig }: { sortConfig?: GoTableSortConfig } = this.localTableConfig;
 
     return sortConfig && sortConfig.column === columnField && sortConfig.direction === direction;
   }
 
-  private setTotalCount() : void {
-    const { totalCount, tableData } = this.localTableConfig;
+  private setTotalCount(): void {
+    const { totalCount, tableData }:
+    {
+      totalCount: number,
+      tableData: any[]
+    } = this.localTableConfig;
 
     this.localTableConfig.totalCount = totalCount !== null ? totalCount : tableData.length;
   }
 
-  private isServerMode() : boolean {
+  private isServerMode(): boolean {
     return this.localTableConfig.dataMode === GoTableDataSource.server;
   }
 
-  private tableChangeOutcome() : void {
+  private tableChangeOutcome(): void {
     this.tableChange.emit(this.localTableConfig);
     if (!this.isServerMode()) {
       this.loadingData = false;
     }
   }
+
+  private determineSelectionMode(): SelectionMode {
+    return this.selectAllChecked ? SelectionMode.deselection : SelectionMode.selection;
+  }
+
+  private isRowInTargeted(row: any): boolean {
```

Comment:
We always know what row will be right? It won't be `any` thing.

Reply:
In this case, `row` is the data for the row in question. Right now, we're treating this as an `any[]`. See: `tableData: any[]`. We could _probably_ make the assumption they'll pass in an `object`, but I'm unsure. I think in this case, we could probably leave this as `any`. Thoughts on that?

Reply:
Should `tableData` be `any` or is it an array of objects? Would it work if it was an array of something else? I'm more so curious than I am suggesting that we change it.

It looks to me like the function uses it like it is an object right? I guess row could technically be an array.